local_planner/src/nodes/planner_functions.cpp

```diff
@@ -117,11 +121,12 @@ void calculateFOV(double h_fov, double v_fov, std::vector<int> &z_FOV_idx,
 }
 
 // Build histogram estimate from reprojected points
-void propagateHistogram(Histogram &polar_histogram_est,
-                        pcl::PointCloud<pcl::PointXYZ> reprojected_points,
-                        std::vector<double> reprojected_points_age,
-                        std::vector<double> reprojected_points_dist,
-                        geometry_msgs::PoseStamped position) {
+void propagateHistogram(
+    Histogram& polar_histogram_est,
+    const pcl::PointCloud<pcl::PointXYZ>& reprojected_points,
+    const std::vector<double>& reprojected_points_age,
+    const std::vector<double>& reprojected_points_dist,
+    geometry_msgs::PoseStamped position) {
```

Comment:
could this be passed by reference? It's quite big actually, 7 doubles plus the header

Reply:
done